core/block.js

```diff
@@ -155,8 +155,21 @@ Blockly.Block = function(workspace, prototypeName, opt_id) {
   // Record initial inline state.
   /** @type {boolean|undefined} */
   this.inputsInlineDefault = this.inputsInline;
+
+  // Fire a create event.
   if (Blockly.Events.isEnabled()) {
-    Blockly.Events.fire(new Blockly.Events.BlockCreate(this));
+    var existingGroup = Blockly.Events.getGroup();
```

Comment:
Creation of a single block didn't need to be in a group before variables started to have IDs and events.  

Creating a block may also create a variable, which fires an event.  If it does, they should be grouped together so that undoing block creation also undoes variable creation.

Sometimes we avoid this problem if the secondary events are deterministic (moving blocks around when changing a proecdure, for instance).  The variable creation is deterministic (for the same XML, on the same workspace) but the ID of the new variable is not--it's a UUID.  So the var create event will need to be fired.

The final piece of the puzzle: the variable creation happens at the first "interaction" with the variable.  In this case, that's when the block is serialized for the BlockCreate event.  Most of the time we avoid this problem, because when loading blocks from XML we turn off events until all of the fields on the blocks have been set.